Installing VS2010 after VS2012 breaks node-gyp

I had node-gyp working fine, then I had to install VS2010 VC++ to work with some other code. Now node-gyp doesn't work from the 2012 command prompt.

I'm on Win8. Please let me know if you have any idea what got changed by the VS2010 install that I need to revert.

Here's the error for VS2012:

```
C:\Program Files (x86)\Microsoft Visual Studio 11.0\VC>cd C:\Code\PTAM

C:\Code\PTAM>node-gyp configure build
gyp info it worked if it ends with ok
gyp info using node-gyp@0.8.4
gyp info using node@0.8.15 | win32 | x64
gyp info spawn python
gyp info spawn args [ 'C:\\Users\\hitsthings\\AppData\\Roaming\\npm\\node_module
s\\node-gyp\\gyp\\gyp',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=auto',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Code\\PTAM\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\hitsthings\\AppData\\Roaming\\npm\\node_module
s\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\hitsthings\\.node-gyp\\0.8.15\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\hitsthings\\.node-gyp\\0.8.15'
,
gyp info spawn args   '-Dmodule_root_dir=C:\\Code\\PTAM',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\Code\\PTAM\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn msbuild
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64' ]
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V110\Microsoft.Cpp.Platform.t
argets(42,5): error MSB8020: The builds tools for Visual Studio 2010 (Platform
Toolset = 'v100') cannot be found. To build using the v100 build tools, either
click the Project menu or right-click the solution, and then select "Update VC+
+ Projects...". Install Visual Studio 2010 to build using the Visual Studio 201
0 build tools. [C:\Code\PTAM\build\ptam-core.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\hitsthings\AppData\Roaming\n
pm\node_modules\node-gyp\lib\build.js:255:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:99:17)
gyp ERR! stack     at Process._handle.onexit (child_process.js:678:10)
gyp ERR! System Windows_NT 6.2.9200
gyp ERR! command "node" "C:\\Users\\hitsthings\\AppData\\Roaming\\npm\\node_modu
les\\node-gyp\\bin\\node-gyp.js" "configure" "build"
gyp ERR! cwd C:\Code\PTAM
gyp ERR! node -v v0.8.15
gyp ERR! node-gyp -v v0.8.4
gyp ERR! not ok

C:\Code\PTAM>
```

If I try from the VS2010 prompt, I get errors that probably mean I need to install Win7 SDK (I'll try that if necessary). I'd prefer to work with VS2012.

```
Setting environment for using Microsoft Visual Studio 2010 x86 tools.

C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC>cd C:\Code\PTAM

C:\Code\PTAM>node-gyp configure build
gyp info it worked if it ends with ok
gyp info using node-gyp@0.8.4
gyp info using node@0.8.15 | win32 | x64
gyp info spawn python
gyp info spawn args [ 'C:\\Users\\hitsthings\\AppData\\Roaming\\npm\\node_module
s\\node-gyp\\gyp\\gyp',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=auto',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Code\\PTAM\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\hitsthings\\AppData\\Roaming\\npm\\node_module
s\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\hitsthings\\.node-gyp\\0.8.15\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\hitsthings\\.node-gyp\\0.8.15'
,
gyp info spawn args   '-Dmodule_root_dir=C:\\Code\\PTAM',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\Code\\PTAM\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn msbuild
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64' ]
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.Cpp.InvalidPlatform
.Targets(23,7): error MSB8007: The Platform for project 'ptam-core.vcxproj' is
invalid.  Platform='x64'. You may be seeing this message because you are trying
 to build a project without a solution file, and have specified a non-default P
latform that doesn't exist for this project. [C:\Code\PTAM\build\ptam-core.vcxp
roj]
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\hitsthings\AppData\Roaming\n
pm\node_modules\node-gyp\lib\build.js:255:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:99:17)
gyp ERR! stack     at Process._handle.onexit (child_process.js:678:10)
gyp ERR! System Windows_NT 6.2.9200
gyp ERR! command "node" "C:\\Users\\hitsthings\\AppData\\Roaming\\npm\\node_modu
les\\node-gyp\\bin\\node-gyp.js" "configure" "build"
gyp ERR! cwd C:\Code\PTAM
gyp ERR! node -v v0.8.15
gyp ERR! node-gyp -v v0.8.4
gyp ERR! not ok

C:\Code\PTAM>
```

--- a/lib/configure.js
+++ b/lib/configure.js
@@ -176,7 +176,7 @@ function configure (gyp, argv, callback) {
            if (code !== 0) {
              checkVCExpress64(cb)
            } else {
-             cb()
+             checkVC2012Express(cb)
            }
          })
   }

